resource_alks_ltk.go

```diff
@@ -75,26 +89,66 @@ func resourceAlksLtkRead(ctx context.Context, d *schema.ResourceData, meta inter
 	providerStruct := meta.(*AlksClient)
 	client := providerStruct.client
 
+	defaultTags := providerStruct.defaultTags
+	ignoreTags := providerStruct.ignoreTags
+
 	// Check if role exists.
 	if d.Id() == "" || d.Id() == "none" {
 		return nil
 	}
 
-	resp, err := client.GetLongTermKey(d.Id())
+	resp, err := client.GetIamUser(d.Id())
 
 	if err != nil {
-		d.SetId("")
-		return nil
+		//If error is 404, UserNotFound, we log it and let terraform decide how to handle it.
+		//All other errors cause a failure
+		if err.StatusCode == 404 {
+			log.Printf("[Error] %s", err.Err)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err.Err)
 	}
 
 	log.Printf("[INFO] alks_ltk.id: %v", d.Id())
 
-	_ = d.Set("iam_username", resp.UserName)
-	_ = d.Set("access_key", resp.AccessKeyID)
+	_ = d.Set("iam_username", resp.User.UserName)
+	_ = d.Set("access_key", resp.User.AccessKey)
+
+	allTags := tagSliceToMap(resp.User.Tags)
+	localTags := removeIgnoredTags(allTags, *ignoreTags)
+
+	if err := d.Set("tags_all", localTags); err != nil {
+		return diag.FromErr(err)
+	}
+
+	userSpecificTags := removeDefaultTags(localTags, defaultTags)
+
+	if err := d.Set("tags", userSpecificTags); err != nil {
+		return diag.FromErr(err)
+	}
 
 	return nil
 }
 
+func resourceAlksLtkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[INFO] ALKS LTK Update")
+
+	// enable partial state mode
+	d.Partial(true)
+
+	if d.HasChange("tags_all") {
+		// try updating enable_alks_access
+		if err := updateUserTags(d, meta); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	d.Partial(false)
+
+	return resourceAlksLtkRead(ctx, d, meta)
+}
+
```

Comment:
I think it's because the tags aren't even returned on the create call through the go client.  They do come back from the ALKS core, but the go client doesn't include them in its return structure.  Thats how it was implemented for roles, so I just stuck to that pattern for users.